policy-test/src/lib.rs

```diff
@@ -61,20 +61,25 @@ pub async fn await_condition<T>(
     ns: &str,
     name: &str,
     cond: impl kube::runtime::wait::Condition<T>,
-) -> Option<T>
+) -> Result<Option<T>, anyhow::Error>
 where
     T: kube::Resource<Scope = kube::core::NamespaceResourceScope>,
     T: serde::Serialize + serde::de::DeserializeOwned + Clone + std::fmt::Debug + Send + 'static,
     T::DynamicType: Default,
 {
     let api = kube::Api::namespaced(client.clone(), ns);
-    time::timeout(
-        time::Duration::from_secs(60),
-        kube::runtime::wait::await_condition(api, name, cond),
-    )
-    .await
-    .expect("condition timed out")
-    .expect("API call failed")
+    tokio::select! {
+        biased;
+        obj  = kube::runtime::wait::await_condition(api, name, cond) => {
```

Comment:
tiny whitespace nit (note that `rustfmt` can't format code inside the `select!` macro, sadly...)
```suggestion
        obj = kube::runtime::wait::await_condition(api, name, cond) => {
```

Reply:
ughh makes sense that it doesn't but I hadn't realised. Thanks!